### ISSUE TITLE
OBW broken when selecting "no thanks" on Benefits step

### ISSUE THREAD

**Issue by jeffstieler** [woocommerce/woocommerce-admin#4467]
**Describe the bug**

A fatal error is encountered when selecting "no thanks" on the OBW JP+WCS step causing the user to get stuck.

This warning is seen in console when entering the benefits step:
```
Property updateProfileItems returned from dispatchMap in useDispatchWithMap must be a function. 
    at WithDispatch(Benefits)
```

This error is seen in console when either benefits step button is pressed (though the "yes please" still succeeds):

```
Uncaught (in promise) TypeError: Cannot read property 'apply' of undefined
in ./client/profile-wizard/steps/benefits/index.js:90
```

**To Reproduce**

1. Ensure JP and WCS are not active
2. Go to OBW
3. Skip to last step
4. Keep existing theme
5. Click "no thanks!"
6. See error

**Expected behavior**

I expect no console errors, and skipping the benefits step to work.

**Additional context**

Introduced in https://github.com/woocommerce/woocommerce-admin/pull/4433 - verified by checking the commit before 0db2c4d99ccbc1289b6e1857a401da38c9c6cd2a.

**Comment by pmcpinto** [comment]
@jeffstieler @joshuatf is this happening in 4.1.1 or in the 4.2 RC? I wasn't able to replicate the issue in 4.1.1

cc @ObliviousHarmony

**Comment by joshuatf** [comment]
@pmcpinto No, this is not included in any release. It was due to a refactor I made to move away from fresh data to wp data in #4433 .

Also related to the long-standing issue I outlined in https://github.com/woocommerce/woocommerce-admin/issues/2959 that makes the whole profiler process so easy to break.  This should be resolved soon as @psealock is working on the last piece in https://github.com/woocommerce/woocommerce-admin/pull/4144 before we can finally tackle #2959.
